core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -139,6 +139,25 @@ public String name() {
     return catalogName;
   }
 
+  private boolean isTableDir(Path path) {
+    Path metadataPath = new Path(path, "metadata");
+    // Only the path which contains metadata is the path for table, otherwise it could be
+    // still a namespace.
+    try {
+      return fs.listStatus(metadataPath, TABLE_FILTER).length >= 1;
+    } catch (IOException e) {
+      return false;
```

Comment:
Same here. Returning `false` is not correct.

Reply:
We use the iceberg HadoopCatalog over Azure ABFS, Users use spark to list the tables in the catalog. However not every user has access to all tables on disk (we use file system access control list). When a user cannot access a folder the listStatus function will throw an IOException.

In my ticket I suggested using a flag to configure the HadoopCatalog to throw errors when unable to list these directories or return false if it is expected that some users will not be able to list all folders/tables.

What do you think about that proposal?

[https://github.com/apache/iceberg/issues/1941](https://github.com/apache/iceberg/issues/1941)

Reply:
I think it makes sense to catch `FileNotFoundException` and return false. What is the exception class that you get when a user can't list path? I would rather use that instead of suppressing any `IOException` if there is a flag set.

It is also unclear how that suppression is set.

Reply:
I've added support for a flag. However I have not yet retrieved the value from the configuration properties.

The exception thrown is this one:

https://hadoop.apache.org/docs/r3.2.0/api/org/apache/hadoop/fs/azurebfs/contracts/exceptions/AbfsRestOperationException.html

IOException
    AzureBlobFileSystemException
        AbfsRestOperationException

Unfortunately the exception thrown is an ABFS specific class.

Reply:
Should the boolean be configurable? How are you setting it?

Reply:
Hey Ryan, sorry for the late reply. Ya I'm thinking it could support a flag passed in as a configuration via the org.apache.hadoop.conf.Configuration

I'm just not sure how to pass values into that configuration object. For example there is already a FILE_IO_IMPL supported. How would I add an additional configuration like FILE_IO_IMPL?

Reply:
Getting the config from catalog properties looks good, but I'd move the property constant as I noted above.

Reply:
this is really brittle design. It will only work for one specific store and contains the assumption that the store always returns the string "AuthorizationPermissionMismatch" on an error. If the abfs store *ever* changed that string everything will break.

* That abfs exception also includes a status code, 403, which is what should really be used as the signal.
* Other stores return AccessDeniedException. Has anyone considered filing a hadoop bug asking for translation of rest exceptions with 403s into those? I am reasonably confident nobody is going reject PRs there.